looper/looper.py

```diff
@@ -877,35 +877,54 @@ def uniqify(seq):
 
 
 
-def check(prj):
+def fetch_flag_files(prj):
     """
-    Check Project status, based on flag files.
-    
-    :param Project prj: Project for which to inquire about status
+    Find all flag file paths for the given project.
+
+    :param Project | AttributeDict prj: full Project or AttributeDict with
+        similar metadata and access/usage pattern
+    :return Mapping[str, list[str]]: collection of filepaths associated with
+        particular flag for samples within the given project
     """
 
-    # TODO: resume here to hook into specific protocols, and if we want
-    # TODO (continued): to just use Python rather than shelling out.
-    from collections import defaultdict
-    flags_by_protocol = defaultdict(list)
+    def _glob_expr(flag):
+        return os.path.join(
+            prj.metadata.results_subdir, "*", "{}.flag".format(flag))
+
+    # TODO: import from pep.
+    flags = ["completed", "running", "failed", "waiting", "partial"]
```

Comment:
the `partial` flag is new to me here

Reply:
I think we should also move the flag list out to a global variable, we may use it elsewhere.